Add event metadata handling

thht opened the pull request:
<!--
Thanks for contributing this pull request (PR).
If this is your first time, make sure to read
[CONTRIBUTING.md](https://github.com/mne-tools/mne-bids/blob/main/CONTRIBUTING.md)
-->

PR Description
--------------

So far, mne-bids only uses `event_id` to handle events. However, we really like and use the event metadata functionality of MNE Python. So, here is an update that you can also use event metadata as an alternative.

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

sappelhoff commented:
Hi @thht, I just had a quick glance and it already looks very clean and reasonable!

Please also follow the steps described here (in particular a changelog item with a high level description of your addition):

https://github.com/mne-tools/mne-bids/blob/main/CONTRIBUTING.md#instructions-for-first-time-contributors

And tag me for a review once you are ready 🙏

thht commented:
@sappelhoff Thanks for the quick reply!

As you can see, I added the authors, citation and changelog information. I also fixed the docs issue.

Failing tests should be due to the bids-validator issue you already reported in #2059.

So, I think, this one is ready for review!

sappelhoff reviewed (approved):
I am very happy with this, thanks a lot!
